.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,28 @@
+name: golangci-lint
+on:
+  pull_request:
+    branches:
+      - master
+
+permissions:
+  contents: read
+
+concurrency:
+  group: '$${{ github.workflow }}-$${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+        with:
+          go-version-file: go.mod
```

Comment:
I'm a little surprised these aren't the default values?

Reply:
Likely, let me check as I'm used to nested/mono repos which do need this (often using matrix).

Reply:
Without the args the action setup `1.20.7` which is the desired GO version.  However the action's logging is a bit unclear on the version selection - ideally it's using the cache which is based on the parsed repo root - so good to validate next time the `go.mod` version is bumped.  Regardless this is just for the lint action, it doesn't impact the actual module.